src/message.rs

```diff
@@ -1375,13 +1377,31 @@ pub async fn delete_msgs(context: &Context, msg_ids: &[MsgId]) -> Result<()> {
                 paramsv![msg.rfc724_mid],
             )
             .await?;
+
+        let logging_xdc_id = context
+            .debug_logging
+            .read()
+            .await
+            .as_ref()
+            .map(|dl| dl.msg_id);
+
+        if let Some(id) = logging_xdc_id {
+            if id == *msg_id {
+                set_debug_logging_xdc(context, None).await?;
+            }
+        }
     }
 
     if !msg_ids.is_empty() {
         context.emit_msgs_changed_without_ids();
 
         // Run housekeeping to delete unused blobs.
-        context.set_config(Config::LastHousekeeping, None).await?;
+        // We need to use set_raw_config() here since with set_config() it
+        // wouldn't compile ("recursion in an `async fn`")
+        context
+            .sql
+            .set_raw_config(Config::LastHousekeeping.as_ref(), None)
+            .await?;
```

Comment:
This change probably isn't needed anymore since you don't change `set_config()`.